src/lib.rs

```diff
@@ -153,11 +182,12 @@ pub fn decode_single_frame_compressed(
     };
     match ret.status {
         0 => unsafe {
-            let slice = slice::from_raw_parts_mut(ret.pixel_data as *mut _, ret.size);
-            Ok(Box::from_raw(slice))
+            let vec = slice::from_raw_parts_mut(ret.pixel_data as *mut _, ret.size).to_vec();
+            c_free_buffer(ret.pixel_data);
```

Comment:
That would work. An alternative is to return a value of a new type encapsulating the allocated data, then freed via the `Drop` impl.

It is a matter of deciding whether this should be handled before releasing a new version or in the next major bump. I imagine that creating a copy of native pixel data from very large multi-frame files could be undesirable.

Reply:
True, this means a couple of copy steps when invoked via dicom-rs. It would change the signature again if this is fixed, so it is probably better to try this in this PR if you don't mind @ingwinlu ?

Reply:
[371fd1d](https://github.com/pevers/gdcm-rs/pull/7/commits/371fd1d15cc5c5deb482a9bf3cdc44ac8d3be320)

please review carefully as I never had to use NonNull before

Reply:
would it actually be better to use UnsafeCell in this case?